sui_programmability/framework/sources/TxContext.move

```diff
@@ -43,19 +46,26 @@ module Sui::TxContext {
     /// Generate a new, globally unique object ID with version 0
     public fun new_id(ctx: &mut TxContext): VersionedID {
         let ids_created = ctx.ids_created;
-        let id = ID::new_versioned_id(fresh_id(*&ctx.tx_hash, ids_created));
+        let id = ID::new_versioned_id(derive_id(*&ctx.tx_hash, ids_created));
         ctx.ids_created = ids_created + 1;
         id
     }
 
+    /// Return the most recent created object ID.
+    public fun get_last_created_object_id(self: &TxContext): ID {
```

Comment:
Should this be `#[test_only]`?

Reply:
Do you think this may be useful API in general?

Reply:
Can this be "racy" in real life scenarios (or even in test scenarios)? Is it possible to use the same context in parallel and not see what the programmer expects to see?

Reply:
I don't think so

Reply:
How does this shake out for safety?

Maybe me still getting up to speed :) 
Like can this be used to fake an object? Or is `VersionedID` the trusted one?

Reply:
`VersionedID` is the trusted one.
I don't think there is safety concern but, let me put this under test only for now until we are 100% certain.